_posts/2016-04-06-Testing-Schemes.markdown

```diff
@@ -0,0 +1,65 @@
+On my fastest computer, we're just under a minute - `Executed 1105 tests, with 1 failure (0 unexpected) in 43.221 (48.201) seconds`  for the whole suite. I think I could probably live with 20 seconds max. So I studied how [AppCode](https://www.jetbrains.com/objc/) handles running tests, and this will be an illustrated guide as to how you can easily run the subset of tests in Xcode based on their techniques.
+
+<!-- more -->
+
+### What are Schemes?
+
+> An Xcode scheme defines a collection of targets to build, a configuration to use when building, and a collection of tests to execute.
+
+> You can have as many schemes as you want, but only one can be active at a time. You can specify whether a scheme should be stored in a project—in which case it’s available in every workspace that includes that project, or in the workspace—in which case it’s available only in that workspace. When you select an active scheme, you also select a run destination (that is, the architecture of the hardware for which the products are built).
+
+To [quote Apple](https://developer.apple.com/library/ios/featuredarticles/XcodeConcepts/Concept-Schemes.html).
+
+### Hatching a Scheme
+
+The Eigen Test Suite is around 50 test classes, these look a bit like this:
+
+![Tests](/images/2016-04-06-Testing-Schemes/tests.png)
+
+We want to be able to say, I want to only run tests for the Fairs section of Eigen, as that is where I will be working for the next few days. To get started, I need to create a new Scheme. You will have seen the schemes before when you click on the Target side of this Targer/Sim button in the top left of Xcode.
+
+![Empty Scheme](/images/2016-04-06-Testing-Schemes/empty_scheme.png)
+
+This is what it looks like on mine, we want to create a new scheme. This brings up a modal window where you need to choose your App's target (this is so you can continue running by pressing `cmd + r`.)
+
+![New Scheme](/images/2016-04-06-Testing-Schemes/new_scheme.png)
+
+I've called mine "Artsy just for Fairs", as I'm the only person who would see it, I can name it whatever I want. Clicking "OK" selects it, and hides the modal. You now need to go back to the target selector, and go to "Edit Schemes ..." to continue though.
+
+![Edit Schemes](/images/2016-04-06-Testing-Schemes/edit_schemes.png)
+
+### Making Amends
+
+OK, click on "Test" in the sidebar, and now you're in the Schemes Test editor. This is where you do the work.
+
+![Empty Edit Schemes](/images/2016-04-06-Testing-Schemes/empty_edit_schemes.png)
+
+You need to hit the "+" in order to add your Test Target to the Scheme
+
+![Test Scheme](/images/2016-04-06-Testing-Schemes/test_scheme.png)
+
+Select and "Add" you Targets. This adds the target, you then need to click the disclosure arrow in order to show all the Test Classes.
+
+_OK, here's some magic_. Hold `alt` and click on the blue tickbox to the right of your test target to turn it off. Then click again, without `alt`. It should then deselect all of the classes.
```

Comment:
I knew that, but me + ash couldn't find another Mac App that does this to show as an example. It's one of those "you only know if you've been a mac dev" things.